packages/react-lowcode/src/codegen/generation/generators/detail/mui-detail-generator.ts

```diff
@@ -1063,4 +1097,146 @@ export default class MuiDetailGenerator implements DetailGenerator {
       return this.createTextFieldComponent(name, text);
     }
   }
+
+
+
+  createInputElementFromTemplate = (property: Property, template: string = ''): string | undefined => {
+    const propName = property.getName();
+    const propType: PropertyType = getPropertyType(property);
+
+    switch (propType) {
+      case PropertyType.string:
+      case PropertyType.numeric: {
+        template = `
```

Comment:
I would like:
* multiple templates (see the issue definition).
* reusable reusable code as posielbe (should be independent from material-ui and grommet)
* prepare for externalized templates

It is suspicious that no code is deleted (the templates are supposed to not only make the solution extensible but also the implementation simpler).